### PR TITLE
ref(ui) Update Issue details chart to echarts 

### PR DESCRIPTION
**Pull request by markstory**
Replace the group release chart with an echarts based implementation. This helps make tooltips more consistent and saves us some code as our echarts code already has robust tooltip handling.

## Before

| Single Series | Environment Filter |
|--------------|--------------------|
| Screen Shot 2020-10-16 at 1 47 13 PM  |  Screen Shot 2020-10-16 at 1 46 54 PM |

## After

| Single Series | Environment Filter |
|--------------|--------------------|
| Screen Shot 2020-10-16 at 4 37 34 PM | Screen Shot 2020-10-16 at 4 38 09 PM |

### CODE REVIEW

File: src/sentry/static/sentry/app/components/group/releaseChart.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import PropTypes from 'prop-types';
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/group/releaseChart.tsx:1]:
```suggestion
```

it looks like that PropTypes is not being used

File: src/sentry/static/sentry/app/components/group/releaseChart.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import React from 'react';
+import styled from '@emotion/styled';
+
+import MiniBarChart from 'app/components/charts/miniBarChart';
+import SentryTypes from 'app/sentryTypes';
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/group/releaseChart.tsx:6]:
```suggestion
```

same here

File: src/sentry/static/sentry/app/components/group/releaseChart.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import {Group, GroupStats, Release} from 'app/types';
+
+type Markers = React.ComponentProps<typeof MiniBarChart>['markers'];
+
+/**
+ * Stats are provided indexed by statsPeriod strings.
+ */
+type StatsGroup = {
+  [key: string]: GroupStats[];
+};
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/group/releaseChart.tsx:20]: maybe 

```suggestion
type StatsGroup = Record<string, GroupStats[]>;
```

File: src/sentry/static/sentry/app/components/group/releaseChart.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      markers.push({
+        name: t('Last seen'),
+        value: lastSeenX,
+        color: theme.green400,
+      });
+    }
+  }
+
+  return (
+    <Container className={className}>
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/group/releaseChart.tsx:107]: I think we usually prefer to name it "Wrapper"

File: src/sentry/static/sentry/app/components/group/releaseChart.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        showTimeInTooltip
+        height={40}
+        series={series}
+        markers={markers}
+      />
+    </Container>
+  );
+}
+
+const Container = styled('div')`
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/group/releaseChart.tsx:120]: shouldn't this emotion style be after the default export?

**Comment by priscilawebdev** [comment]
@mark, please, is the difference in visual snapshots on purpose?

**Comment by markstory** [comment]
> please, is the difference in visual snapshots on purpose? I noticed that the spacing is different from the previous snapshot

It is somewhat on purpose. The old chart relied on positioning the last/first seen markers outside of the chart container in an overflow position. This doesn't work with echarts as it doesn't allow chart elements to overflow the svg container. To ensure that the markers don't get clipped on the sides/bottom, I've spaced the chart in 4px on the left, right and bottom edges, which is the source of the spacing difference.

The markers will also overlap more frequently now in the 30 day chart as there isn't a way to offset markLine components from where they are naturally plotted. The previous chart would always offset the first/last seen markers to avoid this. Unfortunately the `symbolOffset` option for `markLine` components doesn't work. There is an open issue in echarts for that, and if I have time next week I can try to send a fix upstream for it.

**Review by priscilawebdev — APPROVED** [review]

**Pull request merged**